src/main/webapp/WEB-INF/view/conversations.jsp

```diff
@@ -58,35 +60,36 @@
       </form>
 
       <hr/>
-    <% } %>
-
-    <h1>Conversations</h1>
-
-    <%
-    List<Conversation> conversations =
-      (List<Conversation>) request.getAttribute("conversations");
-    if(conversations == null || conversations.isEmpty()){
-    %>
-      <p>Create a conversation to get started.</p>
-    <%
-    }
-    else{
-    %>
-      <ul class="mdl-list">
-    <%
-      for(Conversation conversation : conversations){
-    %>
-
-      <li><a href="/chat/<%= conversation.getTitle() %>">
-        <%= conversation.getTitle() %></a></li> 
+      <h1>Conversations</h1>
 
-    <%
+      <%
+      HashSet<Conversation> conversations =
+        (HashSet<Conversation>) request.getAttribute("conversations");
+      if(conversations == null || conversations.isEmpty()){
+      %>
+        <p>You have no conversations to view.</p>
+      <%
+      } else {
+      %>
+        <ul class="mdl-list">
+      <%
+        for(Conversation conversation : conversations){
```

Comment:
Since we are iterating over the conversations to add them to the page, HashSets aren't very good for that, since they can return stuff in any order. After some googling, I found a "LinkedHashSet" (I think you used them at some point right?), which has the efficiency of HashSet but has pointers to make iteration return the same order every time. So even though I really that you used HashSets, that might be a better option.